internal/host/validator.go

```diff
@@ -1432,3 +1432,40 @@ func (v *validator) printNonOverlappingSubnets(c *validationContext, status Vali
 	}
 	return fmt.Sprintf("Unexpected status %s", status)
 }
+
+func (v *validator) isVSphereDiskUIDEnabled(c *validationContext) ValidationStatus {
+	if c.inventory == nil {
+		return ValidationPending
+	}
+
+	if c.cluster == nil {
+		return ValidationPending
+	}
+
+	if c.cluster.Platform == nil || c.cluster.Platform.Type == nil || *c.cluster.Platform.Type != models.PlatformTypeVsphere {
+		return ValidationSuccess
+	}
+
+	if c.inventory.Disks == nil {
+		return ValidationPending
+	}
+
+	for _, disk := range c.inventory.Disks {
+		if v.hwValidator.IsHDDevice(disk) && !disk.HasUUID {
+			return ValidationFailure
+		}
+	}
+
+	return ValidationSuccess
+}
+
+func (v *validator) printVSphereUUIDEnabled(context *validationContext, status ValidationStatus) string {
+	switch status {
+	case ValidationSuccess:
+		return "VSphere disk.UUID is enabled for this host"
+	case ValidationFailure:
+		return "VSphere disk.UUID isn't enabled for this vm, it's necessary for disks to be mounted properly"
```

Comment:
In the commit message you call it disk.EnableUUID but here it's disk.UUID? which one is it?

Reply:
disk.EnableUUID is the vsphere property name set in the vm level

Reply:
but even if it set for a VM its disks might not have a UUID depend on the disk type so the check should be for every disk

Reply:
misunderstood you concern. It's the same as @romfreiman's

Reply:
Mine was more about host vs disk
Rom seems to be about "Host" vs "VM"